libsolidity/ast/ASTJsonImporter.cpp

```diff
@@ -95,6 +95,20 @@ SourceLocation const ASTJsonImporter::createSourceLocation(Json::Value const& _n
 	return solidity::langutil::parseSourceLocation(_node["src"].asString(), m_sourceNames);
 }
 
+optional<vector<SourceLocation>> ASTJsonImporter::createSourceLocations(Json::Value const& _node) const
+{
+	vector<SourceLocation> locations;
+
+	if (_node.isMember("nameLocations") && _node["nameLocations"].isArray())
+	{
+		for (auto const& val: _node["nameLocations"])
+			locations.emplace_back(langutil::parseSourceLocation(val.asString(), m_sourceNames));
```

Comment:
```suggestion
			locations.emplace_back(createSourceLocation(val, m_sourceNames));
```
Could use the existing method (It enforces an assert that maybe is important...) ?

Reply:
Indeed, we have an array of locations-strings here that need to be parsed directly